parsers/AU-LH.py

```diff
@@ -0,0 +1,200 @@
+        )
+
+    # every other tpspan contains the text "L"
+    even_index_values = values[::2]
+    if not all(val == "L" for val in even_index_values):
+        raise ParserException(
+            PARSER_NAME,
+            "Unexpected parsed values (although normally discarded), website structure changed",
+            zone_key,
+        )
+
+    # the last 4 values are the 4 consumption/production values we care about
+    odd_index_values = values[1::2]
+    cleaned_values = []
+
+    for val in odd_index_values:
+        # Observed units are always kW, this could be a wrong assumption
+        unit_suffix = " kW"
+        if not val.endswith(unit_suffix):
+            raise ParserException(
+                PARSER_NAME,
+                f"Unexpected parsed values, check website structure or electricity units {odd_index_values}",
+                zone_key,
+            )
+
+        # strip suffix, convert to numerical value, convert units to MW
+        parsed_val = float(val.rstrip(unit_suffix)) / 1000
+        num_digits = 10
+        cleaned_values.append(round(parsed_val, num_digits))
+
+    return cleaned_values
+
+
+def fetch_data(zone_key: ZoneKey, session: HTMLSession) -> Dict[str, float]:
+    res = session.get(DATA_URL)
+
+    if not res.status_code == 200:
+        raise ParserException(
+            PARSER_NAME,
+            f"Exception when fetching production error code: {res.status_code}",
+            zone_key,
+        )
+
+    # render the website using request_html (chromium), need a sleep value to wait for render
+    res.html.render(timeout=RENDER_TIMEOUT + 1, sleep=RENDER_TIMEOUT)
+
+    cleaned_values = validate_and_clean_data(zone_key, res)
+
+    value_dict = {
+        "consumption": cleaned_values[0],
+        "solar": cleaned_values[1],
+        "battery": cleaned_values[2],
+        "diesel": cleaned_values[3],
+    }
+
+    # determine the sign of the battery value based on the other values
+    if value_dict["diesel"] + value_dict["solar"] < value_dict["consumption"]:
+        value_dict["battery"] *= -1
+
+    return value_dict
+
+
+def fetch_production(
+    zone_key: ZoneKey,
+    session: HTMLSession = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> Union[List[dict], dict]:
+    if target_datetime is not None:
+        raise ParserException(
+            PARSER_NAME, "This parser is not yet able to parse past dates", zone_key
+        )
+
+    is_new_session = session is None
+    if is_new_session:
+        session = HTMLSession()
+
+    data = fetch_data(zone_key, session)
+
+    production_list = ProductionBreakdownList(logger=logger)
+    production_list.append(
+        zoneKey=zone_key,
+        datetime=datetime.now(timezone.utc),
+        production=ProductionMix(
+            gas=data["diesel"],
+            solar=data["solar"],
+        ),
+        storage=StorageMix(battery=data["battery"]),
```

Comment:
```suggestion
        storage=StorageMix(battery=-data["battery"]),
```
Storage values should be inverted as we count positive as storage charge and negative as storage discharge.

Reply:
I think this is properly considered in the code (line 101 in the new push to the branch), I invert the number if needed in that conditional statement. The battery number can be charge or discharge depending on the time of day.